model/task/task.go

```diff
@@ -2470,83 +2471,103 @@ func (t *Task) Insert() error {
 // into the old_tasks collection. If this is a display task, its execution tasks
 // are also archived.
 func (t *Task) Archive() error {
-	if t.DisplayOnly && len(t.ExecutionTasks) > 0 {
-		execTasks, err := FindAll(db.Query(ByIds(t.ExecutionTasks)))
-		if err != nil {
-			return errors.Wrap(err, "retrieving execution tasks")
-		}
-		if err = ArchiveMany(execTasks); err != nil {
-			return errors.Wrap(err, "archiving execution tasks")
+	if t.DisplayOnly {
+		var task []Task
+		task = append(task, *t)
+		err := ArchiveMany(task)
+		return errors.Wrapf(err, "archiving and updating display task with ID '%s'.", t.Id)
+	} else {
+		// Archiving a single task.
+		if !t.ResetFailedWhenFinished || evergreen.IsFailedTaskStatus(t.Status) {
```

Comment:
This else block covers non-display tasks, non-execution tasks. This was so a restart when failed at a version level could be done with proper archiving of successful results.

If ResetFailedWhenFInished is false (reset all, including successful), the "!" will make it true and it will update it. If it is true, it will only update it if the task is a failed status.